storage/src/main/scala/ch/epfl/bluebrain/nexus/storage/auth/AuthorizationMethod.scala

```diff
@@ -0,0 +1,90 @@
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+/**
+  * Authorization config
+  */
+sealed trait AuthorizationMethod {
+
+  /**
+    * Validates the incoming token
+    */
+  def validate(token: Option[AuthToken]): Either[AuthorizationError, Unit]
+}
+
+object AuthorizationMethod {
+
+  /**
+    * No token/authorization is needed when performing calls
+    */
+  final case object Anonymous extends AuthorizationMethod {
+    override def validate(token: Option[AuthToken]): Either[AuthorizationError, Unit] = Right(())
+  }
+
+  /**
+    * A token matching this realm and username is required and can be validated to the provided audiences and set of
+    * JSON Web Keys
+    */
+  final case class VerifyToken(issuer: String, subject: String, audiences: Option[NonEmptySet[String]], keys: JWKSet)
+      extends AuthorizationMethod {
+    override def validate(token: Option[AuthToken]): Either[AuthorizationError, Unit] = {
+      for {
+        token       <- token.toRight(NoToken)
+        parsedToken <- ParsedToken.fromToken(token).leftMap(InvalidToken)
+        _           <- Either.cond(
+                         issuer == parsedToken.issuer && subject == parsedToken.subject,
+                         (),
+                         UnauthorizedUser(parsedToken.issuer, parsedToken.subject)
+                       )
+        _           <- parsedToken.validate(audiences, keys).leftMap(TokenNotVerified)
+      } yield ()
+    }
+  }
+
+  @nowarn("cat=unused")
+  implicit val authorizationMethodConfigReader: ConfigReader[AuthorizationMethod] = {
+    implicit val jwk: ConfigReader[JWK]                 = ConfigReader.fromStringTry { s => Try(JWK.parse(s)) }
+    implicit val jwkSet: ConfigReader[JWKSet]           = ConfigReader[NonEmptyList[JWK]].map { l => new JWKSet(l.toList.asJava) }
+    implicit val verifyToken: ConfigReader[VerifyToken] = deriveReader[VerifyToken]
+
+    ConfigReader.fromCursor { cursor =>
+      for {
+        obj           <- cursor.asObjectCursor
+        mc            <- obj.atKey("method")
+        discriminator <- ConfigReader[String].from(mc)
+        method        <- discriminator match {
+                           case "anonymous"    => Right(Anonymous)
+                           case "verify-token" => verifyToken.from(obj)
+                           case other          =>
+                             Left(
+                               ConfigReaderFailures(
+                                 ConvertFailure(
+                                   CannotConvert(
+                                     other,
+                                     "string",
+                                     "'method' value must be one of ('anonymous', 'verify-token')"
+                                   ),
+                                   obj
+                                 )
+                               )
+                             )
+                         }
+      } yield method
+    }
```

Comment:
I'm curious what the difference is between this and the method from `Credentials`, I would have thought that the result was the same?